controller/internal/supervisor/iptablesctrl/acls.go

```diff
@@ -293,15 +313,32 @@ func (i *iptables) generateACLRules(cfg *ACLInfo, rule *aclIPset, chain string,
 			iptRules = append(iptRules, rejectRule)
 		}
 
-		if rule.policy.Action&policy.Accept != 0 && (proto == constants.UDPProtoNum || proto == constants.UDPProtoString) {
-			reverseRules = append(reverseRules, []string{
-				appPacketIPTableContext,
-				reverseChain,
-				"-p", proto,
-				"-m", "set", "--match-set", rule.ipset, reverseDirection,
-				"-m", "state", "--state", "ESTABLISHED",
-				"-j", "ACCEPT",
-			})
+		if rule.policy.Action&policy.Accept != 0 {
+			ipRule := []string{}
+
+			// If it is 'all', we add established accept rule with  !tcp.
+			if proto == constants.AllProtoString {
+				ipRule = []string{
+					appPacketIPTableContext,
+					reverseChain,
+					"!", "-p", "tcp",
+					"-m", "set", "--match-set", rule.ipset, reverseDirection,
+					"-m", "state", "--state", "ESTABLISHED",
+					"-j", "ACCEPT",
+				}
+				// If its not TCP, we add established accept rule with given protocol.
+			} else if proto != constants.TCPProtoNum && proto != constants.TCPProtoString {
+				ipRule = []string{
+					appPacketIPTableContext,
+					reverseChain,
+					"-p", proto,
+					"-m", "set", "--match-set", rule.ipset, reverseDirection,
+					"-m", "state", "--state", "ESTABLISHED",
+					"-j", "ACCEPT",
+				}
+			}
+
```

Comment:
Doesn't this affect UDP As well ?

Reply:
If you look at the diff, it used to be just UDP (yes, we need to accept it). This was a bug as icmp (or any other protocol) was being dropped on the net side. Now with this we accept established connections of all protocols but tcp.

Reply:
` else if proto != constants.TCPProtoNum && proto != constants.TCPProtoString `